acorn/src/statement.js

```diff
@@ -568,6 +581,10 @@ pp.parseFunctionParams = function(node) {
 // Parse a class declaration or literal (depending on the
 // `isStatement` parameter).
 
+pp.shouldCheckClassConstructorDuplicate = function() {
```

Comment:
How about we use `raiseRecoverable` for the duplicate constructor error, and your plugin overrides that and drops the error when appropriate?

Reply:
Thats a good idea to handle error with `raiseRecoverable `, thx!